src/client.js

```diff
@@ -115,15 +115,14 @@ export default class Client {
   //
 
   /**
-   * Initiate connection to the IMAP server
+   * Initiate connection and login to the IMAP server
    *
    * @returns {Promise} Promise when login procedure is complete
    */
   async connect () {
     try {
-      await this._openConnection()
+      await this.openConnection()
       this._changeState(STATE_NOT_AUTHENTICATED)
```

Comment:
looks like
`      this._changeState(STATE_NOT_AUTHENTICATED)`
should be inside
`this.openConnection()`
as that is now a public facing function

Reply:
yes, makes sense. I moved it